pkg/controllers/monitoring/monitoring-stack/controller.go

```diff
@@ -61,7 +66,7 @@ type Options struct {
 //+kubebuilder:rbac:groups=monitoring.rhobs,resources=monitoringstacks/status,verbs=get;update
 
 // RBAC for managing Prometheus Operator CRs
-//+kubebuilder:rbac:groups=monitoring.coreos.com,resources=alertmanagers;prometheuses;servicemonitors,verbs=list;watch;create;update;delete;patch
+//+kubebuilder:rbac:groups=monitoring.coreos.com,resources=alertmanagers;prometheuses;servicemonitors,verbs=list;watch;create;update;delete;patch;get
```

Comment:
Having to add `get` probably indicates that we are skipping the cache which I think we shouldn't and has in the past led to hard to track reconcile errors due to race conditions caused by mismatch of objects in cache and the one returned by the server. Since the controller already watches prometheus, can't we use the dynamic client to `Get` prometheus from the Cache itself?

Reply:
Ahh interesting. The dynamic client you're talking about is the `k8sClient` available in the controller (obtained from the manager) right? I can try that.